src/js/controller/model.js

```diff
@@ -102,7 +102,7 @@ define([
 
                     /* falls through */
                 case events.JWPLAYER_MEDIA_META:
-                    var duration = data.duration;
+                    var duration = data.duration || 0;
```

Comment:
It looks like we never tried to set the duration if ```data.duration``` is not a number. This changes the behavior, and probably fires an additional metadata event with a duration update. Can you explain which part passes ```NaN``` to the duration?